src/main/java/org/skife/jdbi/v2/ClasspathStatementLocator.java

```diff
@@ -74,20 +75,32 @@ public static boolean looksLikeSql(String sql)
     @SuppressFBWarnings("DM_STRING_CTOR")
     public String locate(String name, StatementContext ctx)
     {
-        final String cache_key;
+        final Object cache_key;
+        boolean isSqlObjectMethod = false;
         if (ctx.getSqlObjectType() != null) {
-            cache_key = '/' + mungify(ctx.getSqlObjectType().getName() + '.' + name) + ".sql";
+            if (ctx.getSqlObjectMethod() != null) {
+                isSqlObjectMethod = true;
+                cache_key = ctx.getSqlObjectMethod();
+            } else {
+                cache_key = '/' + mungify(ctx.getSqlObjectType().getName() + '.' + name) + ".sql";
+            }
         }
         else {
             cache_key = name;
         }
 
-        if (found.containsKey(cache_key)) {
-            return found.get(cache_key);
+        String cached = found.get(cache_key);
+        if (cached != null) {
+            return cached;
         }
 
+
         if (looksLikeSql(name)) {
             // No need to cache individual SQL statements that don't cause us to search the classpath
+            // But for static SQL object methods caching decreases GC pressure and not threatens memory leaks.
+            if (isSqlObjectMethod) {
+                found.put(cache_key, name);
```

Comment:
I don't see how this is improving the situation much.  You still have to assemble the cache key, which is the "expensive" portion you are trying to avoid -- filling the cache with these saves you the `looksLikeSql` call but nothing else as far as I can see?